pkg/operator/controller/credentials-secret/credentials_secret.go

```diff
@@ -150,5 +165,5 @@ func (r *reconciler) updateCredentialsSecret(ctx context.Context, current, desir
 // the secrets should be considered equal for the purpose of determining
 // whether an update is necessary, false otherwise.
 func secretsEqual(a, b *corev1.Secret) bool {
-	return reflect.DeepEqual(a.Data, b.Data)
+	return reflect.DeepEqual(a.Data, b.Data) && reflect.DeepEqual(a.StringData, b.StringData)
```

Comment:
The ` desiredCredentialsSecret` function should return all data in `.Data` field, no need to add something in `.StringData`.

Reply:
Removed